src/main/java/Duke.java

```diff
@@ -1,84 +1,66 @@
+        } catch (IOException e) {
+            Error.displayFileLoadError();
         }
-        return taskName.trim();
-    }
-
-    public static void greet() {
-        Display.printSeparatingLine();
-        System.out.println("Hello! I'm Duke");
-        System.out.println("What can I do for you?\n");
     }
 
-    public static void goodbye() {
-        Display.printSeparatingLine();
-        System.out.println("Bye. Hope to see you again soon!\n");
-        Display.printSeparatingLine();
+    /** Runs the program till the user terminates it. */
+    public void run() {
+        interact();
+        Display.displayGoodbyes();
     }
 
-    public static void interact() {
-        String line;
-        Scanner in = new Scanner(System.in);
-        TaskManager taskManager = new TaskManager();
-        Boolean isStillInteracting = true;
-        while (isStillInteracting) {
-            line = in.nextLine();
-            String[] words = line.trim().split(SEPARATOR);
-
-            switch (words[COMMAND_INDEX]) {
-            case COMMAND_LIST_TASK:
-                taskManager.listTask();
-                break;
-            case COMMAND_MARK_DONE:
-                int taskNumber = Integer.parseInt(words[TASK_INDEX]) - 1;
-                taskManager.markTaskAsCompleted(taskNumber);
-                break;
-            case COMMAND_EXIT_PROGRAM:
-                isStillInteracting = false;
-                break;
-            case COMMAND_TODO_TASK:
-                taskManager.addTodoTask(getTaskDetails(words));
-                break;
-            case COMMAND_DEADLINE_TASK:
-                taskManager.addDeadlineTask(getTaskDetails(words));
-                break;
-            case COMMAND_EVENT_TASK:
-                taskManager.addEventTask(getTaskDetails(words));
-                break;
-            default:
-                Error.displayInvalidCommandError();
-                break;
+    /**
+     * Interacts with user by waiting for command line inputs.
+     * It will continue to interact with the user till the program is terminated by a user command.
+     * The command line inputs are parsed accordingly and the appropriate response would be displayed to the user.
+     * The tasks saved in storage would be updated after every command.
+     */
+    public void interact() {
+        Command command;
+        do {
+            String userInput = InputParser.getUserCommand(inputScanner);
+            String[] commandComponents = InputParser.getCommandComponents(userInput);
+            command = InputParser.parseCommand(commandComponents, taskManager);
+            command.setCommandComponents(commandComponents);
+            try {
+                storage.updateFileData(taskManager.getAllTasks());
+            } catch (IOException e) {
+                Error.displayFileUpdateError();
             }
-        }
+        } while (!command.executeCommand().equals(ExitCommand.COMMAND_WORD));
```

Comment:
You may want to introduce a new boolean variable to make this line easier to understand.